### PR TITLE
[V2 API] Add more observation golden tests to match V1

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: internal/server/v2/observation/golden/contained_in_2015_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+				continue
+			}
+			if diff := cmp.Diff(resp, &expected, protocmp.Transform()); diff != "" {
+				t.Errorf("payload got diff: %v", diff)
+				continue
+			}
+		}
+	}
+	if err := test.TestDriver(
+		"Collection",
```
Comment (Spaceenter) [inline internal/server/v2/observation/golden/contained_in_2015_test.go:86]: Update name here?

File: internal/server/v2/observation/golden/contained_in_all_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+				continue
+			}
+			if diff := cmp.Diff(resp, &expected, protocmp.Transform()); diff != "" {
+				t.Errorf("payload got diff: %v", diff)
+				continue
+			}
+		}
+	}
+	if err := test.TestDriver(
+		"Collection",
```
Comment (Spaceenter) [inline internal/server/v2/observation/golden/contained_in_all_test.go:104]: Update name here?

**Review by Spaceenter — APPROVED** [review]

File: internal/server/v2/observation/golden/contained_in_2015_test.go (hunk 1 above)
Comment (shifucun) [reply internal/server/v2/observation/golden/contained_in_2015_test.go:86]: ok, i will update in next PR

File: internal/server/v2/observation/golden/contained_in_all_test.go (hunk 2 above)
Comment (shifucun) [reply internal/server/v2/observation/golden/contained_in_all_test.go:104]: will update in next PR

**Pull request merged**